pkg/controllers/user/alert/configsyncer/configsyncer.go

```diff
@@ -542,6 +560,112 @@ func (d *ConfigSyncer) addRecipients(notifiers []*v3.Notifier, receiver *alertco
 
 				receiver.WebhookConfigs = append(receiver.WebhookConfigs, webhook)
 				receiverExist = true
+
+			} else if notifier.Spec.DingtalkConfig != nil {
+				d.enableWebHookReceiver()
+				webhookSecreteName, altermanagerAppNamespace := monitorutil.SecretWebhook()
+				secretClient := d.secretsGetter.Secrets(altermanagerAppNamespace)
+				configSecret, err := secretClient.Get(webhookSecreteName, metav1.GetOptions{})
+				if err != nil {
+					logrus.Errorf("Failed to get secret : %v", err)
+				}
+
+				oldConfig := configSecret.Data["config.yaml"]
+
+				config := Config{
+					Providers: make(map[string]*Provider),
+					Receivers: make(map[string]*Receiver),
+				}
+				err = yaml.Unmarshal(oldConfig, &config)
+				if err != nil {
+					logrus.Errorf("Error yaml Unmarshal config : %v", err)
+				}
+
+				config.Providers[DingTalk+"-"+r.NotifierName] = &Provider{
+					WebHookURL: notifier.Spec.DingtalkConfig.URL,
+					Secret:     notifier.Spec.DingtalkConfig.Secret,
+				}
+
+				if notifier.Spec.DingtalkConfig.HTTPClientConfig != nil {
+					config.Providers[DingTalk+"-"+r.NotifierName].ProxyURL = notifier.Spec.DingtalkConfig.HTTPClientConfig.ProxyURL
+				}
+
+				config.Receivers[r.NotifierName] = &Receiver{
+					Provider: DingTalk + "-" + r.NotifierName,
+				}
+
+				data, err := yaml.Marshal(config)
+				if err != nil {
+					logrus.Errorf("Error yaml Marshal config : %v", err)
+				}
+
+				configSecret.Data["config.yaml"] = data
+				_, err = secretClient.Update(configSecret)
```

Comment:
This code block looks duplicate with the next condition. Extract a standalone function